src/vs/workbench/contrib/terminal/browser/xterm/quickFixAddon.ts

```diff
@@ -331,3 +338,61 @@ registerThemingParticipant((theme: IColorTheme, collector: ICssStyleCollector) =
 		collector.addRule(`.${DecorationSelector.CommandDecoration}.${DecorationSelector.QuickFix} { background-color: ${backgroundColor.toString()}; } `);
 	}
 });
+
+export function convertExtensionQuickFixOptions(quickFix: IExtensionTerminalQuickFix): ITerminalQuickFixOptions {
+	const type = quickFix.commandToRun ? 'command' : quickFix.linkToOpen ? 'opener' : undefined;
+	const options = {
+		id: quickFix.id,
+		commandLineMatcher: quickFix.commandLineMatcher,
+		outputMatcher: quickFix.outputMatcher,
+		type,
+		getQuickFixes: type === 'command' ? (matchResult: TerminalQuickFixMatchResult) => {
+			const matches = matchResult.outputMatch;
+			const commandToRun = quickFix.commandToRun;
+			if (!matches || !commandToRun) {
+				return;
+			}
+			const groups = matches.groups;
+			if (!groups) {
+				return;
+			}
+			for (const [key, value] of Object.entries(groups)) {
+				const varToResolve = '${group:' + `${key}` + '}';
```

Comment:
to do - does this work if two groups are provided? don't think so atm, but should